mdal/frmts/mdal_ugrid.cpp

```diff
@@ -74,36 +71,97 @@ MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( )
   size_t count;
   int ncid;
 
-  mMesh1dName = findMeshName( 1, true ); // optional, may not be present
-  mMesh2dName = findMeshName( 2, false ); // force
+  mAllMeshNames = findMeshesNames();
 
-  // 2D Mesh
+  if ( mAllMeshNames.empty() )
```

Comment:
Where is the logic that checks that the mMeshName requested from the user is used?

Reply:
It is not there yet

Reply:
Up